cppapi/server/fwdattrdesc.cpp

```diff
@@ -180,10 +181,15 @@ bool FwdAttr::validate_fwd_att(vector<AttrProperty> &prop_list,const string &dev
 	}
 	catch (...) {}
 
-	if (root_att_db_defined == true)
-		full_root_att = root_att_db;
-    else
-        full_root_att = RootAttNotDef;
+	//check if full_root_att is already set
+	is_full_root_att_set = full_root_att.size()!=0 && full_root_att.compare(RootAttNotDef)!=0;
+
+	if(!is_full_root_att_set) {
```

Comment:
This change is introducing a problem when the server is exporting several devices having a standard forwarded attribute (defined with POGO) in their class. If these forwarded attributes are defined to point to different root attributes (since pointing to the same root attribute in the same process is not supported by Tango), only the first created device works as expected. The other devices do not manage to create the forwarded attribute and report the following error in their status:
The device is in ALARM state.```

> 
> Forwarded attribute TestFwdAttr is not correctly configured! Root attribute name = tango://mytangohost:10000/my/root/device/myattr
> You can update it using the Jive tool
> Error: Root attribute already used in this device server process for attribute my/test/forward1/TestFwdAttr

This problem is mainly due to how Tango is managing dynamic attributes. The dynamic attributes are added to the class attributes list when they are created and if an attribute with the same name already exists in the current class attributes list, the already created Attr object is re-used.
So with the current code from this PR, when the standard forwarded attribute for the second device is created, DeviceImpl constructor will cast the Attr object which was already created for the first device and will call the validate_fwd_att() method but on the Attr object which was created for the first device. In this case, full_root_att is not empty and is pointing to the root attribute defined for the first device., leading to this error.